Mesh_3/include/CGAL/Mesh_criteria_3.h

```diff
@@ -93,22 +68,22 @@ class Mesh_criteria_3_impl
   // This template constructor is not instantiated when named parameters
   // are not used, so Facet_criteria and Cell_criteria construction from FT
   // is not a problem
-  template <class ArgumentPack>
-  Mesh_criteria_3_impl(const ArgumentPack& args)
-    : edge_criteria_(args[parameters::edge_size
-                          | args[parameters::edge_sizing_field
-                                 | args[parameters::sizing_field | FT(DBL_MAX)] ] ])
-    , facet_criteria_(args[parameters::facet_angle | FT(0)],
-                      args[parameters::facet_size
-                           | args[parameters::facet_sizing_field
-                                  | args[parameters::sizing_field | FT(0)] ] ],
-                      args[parameters::facet_distance | FT(0)],
-                      args[parameters::facet_topology | CGAL::FACET_VERTICES_ON_SURFACE])
-    , cell_criteria_(args[parameters::cell_radius_edge_ratio
-                          | args[parameters::cell_radius_edge | FT(0)] ],
-                     args[parameters::cell_size
-                          | args[parameters::cell_sizing_field
-                                 | args[parameters::sizing_field | FT(0)] ] ])
+  template <typename CGAL_NP_TEMPLATE_PARAMETERS>
+  Mesh_criteria_3_impl(const CGAL_NP_CLASS& np)
+    :edge_criteria_(parameters::choose_parameter(parameters::get_parameter(np, internal_np::edge_size_param),
+                                                 parameters::choose_parameter(parameters::get_parameter(np, internal_np::edge_sizing_field_param),
```

Comment:
Given that this can be a field, should it be a ref (same for other fields)?

Reply:
if this can indeed contain something like an AABB-tree it should indeed be taken by ref. @lrineau @janetournois do you agree that it should always be a ref or can it sometime be a copy?

Reply:
I can't think of a reason why it would need to be copied, so I agree it should be a taken as a reference

Reply:
It is supposed to be passed by reference to const. It is the way is was documented in CGAL-5.5, as least here:
  - https://doc.cgal.org/5.5/Mesh_3/classCGAL_1_1Mesh__edge__criteria__3.html#aa5da9d0dc1017155ffd6600e03841c30
  - https://doc.cgal.org/5.5/Mesh_3/classCGAL_1_1Mesh__facet__criteria__3.html#a77f4b4b8646e0cf549cdce4f8684ec4d
  - https://doc.cgal.org/5.5/Mesh_3/classCGAL_1_1Mesh__cell__criteria__3.html#a63e0a33fd1656210d260868f5fa176bc

Reply:
I updated the code for the 5 sizing fields and a (const) reference is now required.